zzk/service/hoststate.go

```diff
@@ -79,20 +93,38 @@ func (l *HostStateListener) Ready() error {
 	return nil
 }
 
-// Done removes the ephemeral node from the host registry
+// Done implements zzk.Listener
 func (l *HostStateListener) Done() {
 }
 
 // PostProcess implements zzk.Listener
-func (l *HostStateListener) PostProcess(p map[string]struct{}) {}
+// This is always called after all threads have been spawned
+func (l *HostStateListener) PostProcess(p map[string]struct{}) {
+	// We are running all of the containers we are supposed to, now
```

Comment:
Should this function be under a single lock?  Seems like you can hit some map inconsistencies